gwtp-core/gwtp-mvp-client/src/main/java/com/gwtplatform/mvp/client/PresenterWidget.java

```diff
@@ -225,6 +226,10 @@ public void fireEvent(GwtEvent<?> event) {
         getEventBus().fireEventFromSource(event, this);
```

Comment:
change:
`getEventBus().fireEventFromSource(event, this);`
to:
`fireEvent((Event<?>) event);`

Then any future changes to fire event can be made in the one method.